pyfakefs/fake_filesystem.py

```diff
@@ -262,21 +271,30 @@ def __init__(
 
     @property
     def is_linux(self) -> bool:
+        """Returns `True` in a real or faked Linux file system."""
         return not self.is_windows_fs and not self.is_macos
 
     @property
     def is_windows_fs(self) -> bool:
-        return self._is_windows_fs
+        """Returns `True` in a real or faked Windows file system."""
+        return (
+            self.fs_type == FSType.WINDOWS
+            or self.fs_type == FSType.DEFAULT
+            and self._is_windows_fs
```

Comment:
Oh boy, I _strongly_ recommend against combining `or` and `and` operators without parentheses here. This is unlikely to work the way people read it.

[Python's operator precedence rules](https://docs.python.org/3/reference/expressions.html#operator-precedence) make it clear that the `and` will be evaluated first, followed by the `or` making this code equivalent to:

```python
self.fs_type == FSType.WINDOWS
or (self.fs_type == FSType.DEFAULT and self._is_windows_fs)
```

I recommend adding grouping parentheses to confirm to readers that this is written -- and will evaluate -- as intended, even if you've written this perfectly and know that it will evaluate as intended.